zenoh-flow/src/runtime/dataflow/instance/builtin/zenoh.rs

```diff
@@ -348,23 +348,43 @@ impl<'a> Sink for ZenohSink<'a> {
                     default
                 };
 
-                let shm_elem_size = get_or_default(
-                    &configuration,
-                    KEY_SHM_ELEM_SIZE,
-                    *context.shared_memory_element_size() as u64,
-                ) as usize;
-                let shm_elem_count = get_or_default(
-                    &configuration,
-                    KEY_SHM_TOTAL_ELEMENTS,
-                    *context.shared_memory_elements() as u64,
-                ) as usize;
-                let shm_backoff = get_or_default(
-                    &configuration,
-                    KEY_SHM_BACKOFF,
-                    *context.shared_memory_backoff(),
-                );
-
-                let shm_size = shm_elem_size * shm_elem_count;
+                let (shm_element_size, shm_backoff, shm_manager) =
+                    match context.shared_memory_enabled() {
+                        true => {
```

Comment:
A small internet search seems to indicate that it is a bit more idiomatic to use an `if` block for a boolean.
I'd suggest:

```rust
let mut shm_element_size = 0;
let mut shm_backoff = 0;
let mut shm_manager = None;

if context.shared_memory_enabled() {
// …
}
```
That way it's harder to loose track of which default value corresponds to what (compared to `false => (0, 0, None)`).